### PR TITLE
Use atoms for all CAF message handlers

### PR DESCRIPTION
**Pull request by patszt**
All CAF message handlers will have atom as a first argument.
This should help with code navigation

### :memo: Checklist

- [x] All user-facing changes have changelog entries.
- [x] The changes are reflected on [vast.io](https://vast.io), if necessary.
- [x] The PR description contains instructions for the reviewer, if necessary.

### CODE REVIEW

File: libvast/include/vast/system/actors.hpp [hunk 1]
```diff
@@ -143,7 +143,7 @@ using evaluator_actor = typed_actor_fwd<
 /// The INDEXER actor interface.
```
Comment (lava) [inline libvast/include/vast/system/actors.hpp:143]: GitHub won't let me comment outside of the current diff, but the `receiver_actor` further up in this file still introduces a message handler without an atom:

```
template <class T, class... Ts>
using receiver_actor = typename typed_actor_fwd<
  // Add a new source.
typename caf::reacts_to<T, Ts...>>::unwrap;
```

**Review by lava — APPROVED** [review]
Great to finally see this completed :tada:

File: libvast/include/vast/system/actors.hpp (hunk 1 above)
Comment (patszt) [reply libvast/include/vast/system/actors.hpp:143]: Hmm should it be enforced for the receiver_actor to have an atom? Or the users for receiver_actor should pass the atom?
Any suggestion for atom example for our use cases?

Comment (lava) [reply libvast/include/vast/system/actors.hpp:143]: I'm not sure I understand the distinction between the two cases, I was thinking of something like this:

```
template <class T, class... Ts>
using receiver_actor = typename typed_actor_fwd<
  // Add a new source.
typename caf::reacts_to<atom::receive, T, Ts...>>::unwrap;
```

Comment (patszt) [reply libvast/include/vast/system/actors.hpp:143]: Perfect, i was thinking about the same :)

**Pull request merged**